optuna/samplers/_tpe/sampler.py

```diff
@@ -344,13 +344,16 @@ def _sample_categorical_index(self, distribution, below, above):
         below = list(map(int, below))
         above = list(map(int, above))
         upper = len(choices)
-        size = (self._n_ei_candidates,)
 
         weights_below = self._weights(len(below))
         counts_below = np.bincount(below, minlength=upper, weights=weights_below)
         weighted_below = counts_below + self._prior_weight
         weighted_below /= weighted_below.sum()
-        samples_below = self._sample_from_categorical_dist(weighted_below, size)
+        if self._n_ei_candidates >= upper:
+            samples_below = np.arange(upper)
```

Comment:
I think this changes the logic to guarantees that all the choices will be included. Should confirm the performance with benchmarks, or was it covered in the Goptuna benchmarks?